client/protocols/guidance.py

```diff
@@ -0,0 +1,178 @@
+from client.protocols.common import models
+from client.utils import auth
+from urllib.request import urlopen
+import json
+import os
+import sys
+
+TGSERVER = "http://tg-server.app.cs61a.org/"
+LOCAL_TG = "tests/tg.ok_tg"
+OK_GUIDANCE = "tests/.ok_guidance"
+
+lambda_string_key_to_func = {
+    'none': lambda info, strMisU: None, 
+    'ki': lambda info, strMisU: info['ki'],
+    'reteach': lambda info, strMisU: info['dictMisU2ReteachMsg'].get(strMisU)
+}
+  
+# If student forces guidance messages to show, we will assign treatment group number below
+guidance_flag_dict = [1]
+
+class GuidanceProtocol(models.Protocol):
+    def __init__(self,working_dir,debug = -1):
+        self.tg_id = -1
+        self.msg_id = -1
+        self.misU_count = {}
+        self.working_dir = working_dir
```

Comment:
Not sure what `working_dir` means. Perhaps `current_directory` or `current_working_directory`?